### PR TITLE
Updating map change event documentation with interface listeners

### PR DESCRIPTION
**Pull request by langsmith**
Resolves https://github.com/mapbox/android-docs/issues/914 by updating the writing and code about map change events on https://docs.mapbox.com/android/maps/overview/events/#map-change-events.

This pr adds info about the fact that there are now individual interface listeners/callbacks for every type of map change event.

screencapture-localhost-8080-android-maps-overview-events-2019-04-22-14_17_46

### CODE REVIEW

File: src/pages/maps/overview/events.md [hunk 1]
```diff
@@ -261,11 +261,36 @@ marker.icon
...
+	@Override
+	public void onDidFinishRenderingMap(boolean fully) {
+	    
+	    
+	}
+});
+`}
+
+kotlin={`
+mapView?.addOnDidFinishRenderingMapListener {
```
Comment (tobrun) [inline src/pages/maps/overview/events.md]: when using kotlin, we should assume the developer using kotlin android extensions, we should remove the `?`

File: src/pages/maps/overview/events.md [hunk 2]
```diff
@@ -261,11 +261,36 @@ marker.icon
...
+`}
+
+/>
+}}
+
+You can also implement the specific change event interface and override its method. The interface callback is then invoked when a new map change occurs.
+
+The lifecycle events below are listed in the order in which they occur during a `MapView`'s life.
 
 | MapChange Constants                       | Description |
```
Comment (tobrun) [inline src/pages/maps/overview/events.md]: MapChange Constants do not exist anymore

File: src/pages/maps/overview/events.md [hunk 3]
```diff
@@ -299,37 +324,32 @@ Additional change events that are not part of the standard map lifecycle:
...
+	// The map is now ready for other changes
+	
+	}
+});
 `}
 
 kotlin={`
-override fun onMapChanged(change: Int) {
-        if (change == MapView.DID_FINISH_LOADING_STYLE) {
+mapView?.addOnDidFinishLoadingStyleListener {
```
Comment (tobrun) [inline src/pages/maps/overview/events.md]: `?`

File: src/pages/maps/overview/events.md (hunk 1 above)
Comment (langsmith) [reply src/pages/maps/overview/events.md]: Removed

File: src/pages/maps/overview/events.md (hunk 3 above)
Comment (langsmith) [reply src/pages/maps/overview/events.md]: Removed

File: src/pages/maps/overview/events.md (hunk 2 above)
Comment (langsmith) [reply src/pages/maps/overview/events.md]: Fixed 
Screen Shot 2019-04-23 at 10 53 37 AM

File: src/pages/maps/overview/events.md [hunk 4]
```diff
@@ -261,75 +261,91 @@ marker.icon
...
+| `addOnWillStartRenderingFrameListener `                | This event is triggered when the map will start rendering a frame. |
+| `OnDidFinishRenderingFrameListener `                | This event is triggered when the map finished rendering a frame. |
+| `OnDidFinishLoadingMapListener`                    | This is triggered when the map has successfully loaded a new map style. |
+| `addOnWillStartRenderingFrameListener`                | This event is triggered when the map will start rendering a frame. |
+| `OnDidFinishRenderingMapListener`   | This event is triggered when the map is fully rendered. |
+
+
+Additional change events that are not part of the standard `MapView` lifecycle:
 
 | MapChange Constants                       | Description |
```
Comment (tobrun) [inline src/pages/maps/overview/events.md]: Nit using word `Constants`

File: src/pages/maps/overview/events.md [hunk 5]
```diff
@@ -261,75 +261,91 @@ marker.icon
...
+The `MapView` goes through a series of lifecycle events while building/changing the map. The Maps SDK provides many different change listener interfaces to tell you when a specific map event has occurred.
 
-Instead of adding the listener through the `MapboxMap` object, the listeners added using the `MapView`, `mapView.addOnMapChangedListener(OnMapChangedListener());`. You can also implement the `MapView.OnMapChangedListener` interface and override the `onMapChanged()` method. When a new map change occurs, the `onMapChanged()` callback is invoked. This provides the constant, as an integer, as the parameter which you can then use to match up with one of the constants listed in the table below.
+Instead of adding a listener to the `MapboxMap` object, each listener is added to a `MapView` object. For example `OnDidFinishRenderingMapListener` and `addOnDidFinishLoadingMapListener`.
 
-The lifecycle events below are listed in the order in which they occur during the loading of a `MapView`.
+{{
+<CodeLanguageToggle id="window-click-listener-events" />
```
Comment (colleenmcginnis) [inline src/pages/maps/overview/events.md]: `id` needs to be unique. Currently the same as line 309 causing: 

image

File: src/pages/maps/overview/events.md [hunk 6]
```diff
@@ -261,75 +261,91 @@ marker.icon
...
+mapView.addOnDidFinishRenderingMapListener(new MapView.OnDidFinishRenderingMapListener() {
+	@Override
+	public void onDidFinishRenderingMap(boolean fully) {
+	    
+	    
+	}
+});
+`}
 
+kotlin={`
```
Comment (colleenmcginnis) [inline src/pages/maps/overview/events.md]: I still don't know how kotlin works... is this the intended snippet? (It just seemed a little light to me.)

image

File: src/pages/maps/overview/events.md [hunk 7]
```diff
@@ -261,75 +261,91 @@ marker.icon
 
 ## Map change events
 
-The map view goes through a series of events while building/changing the map. The `OnMapChangedListener` provided can be used to tell you when one or multiple events occur you're interested in. This includes knowing when the map starts and finishes loading the map style or when frames are finished rendering.
+The `MapView` goes through a series of lifecycle events while building/changing the map. The Maps SDK provides many different change listener interfaces to tell you when a specific map event has occurred.
```
Comment (colleenmcginnis) [inline src/pages/maps/overview/events.md]: `building/changing` --> `building and changing`

File: src/pages/maps/overview/events.md (hunk 6 above)
Comment (colleenmcginnis) [reply src/pages/maps/overview/events.md]: Maybe a commented line like https://github.com/mapbox/android-docs/pull/934/files#diff-8d39f7e6e2cc77868577a585b785c1acR345 ?

File: src/pages/maps/overview/events.md [hunk 8]
```diff
@@ -261,75 +261,91 @@ marker.icon
...
+| `OnWillStartLoadingMapListener`                    | This event is triggered when the map is about to start loading a new map style. |
+| `OnWillStartRenderingMapListener`                  | This event is triggered when the map will start rendering the map. |
+| `addOnWillStartRenderingFrameListener`                | This event is triggered when the map will start rendering a frame. |
+| `OnDidFinishRenderingFrameListener`                | This event is triggered when the map finished rendering a frame. |
+| `OnCameraWillChangeListener`                        | This event is triggered whenever the displayed map region is about to change without an animation. |
+| `OnCameraDidChangeListener`                         | This event is triggered whenever the displayed map region finished changing without an animation. |
+| `OnDidFinishLoadingStyleListener`                  | Triggered when a style has finished loading. |
+| `OnDidFinishRenderingFrameListener`                | This event is triggered when the map finished rendering a frame. |
+| `OnSourceChangedListener`                         | Triggered when a source attribution changes. |
```
Comment (colleenmcginnis) [inline src/pages/maps/overview/events.md]: This phrase seems strange to me: 

>a source attribution

I'm wondering why you're not just using `a source` in this context.

File: src/pages/maps/overview/events.md (hunk 5 above)
Comment (langsmith) [reply src/pages/maps/overview/events.md]: fixed

File: src/pages/maps/overview/events.md (hunk 7 above)
Comment (langsmith) [reply src/pages/maps/overview/events.md]: fixed

File: src/pages/maps/overview/events.md (hunk 8 above)
Comment (langsmith) [reply src/pages/maps/overview/events.md]: Fixed

File: src/pages/maps/overview/events.md (hunk 6 above)
Comment (langsmith) [reply src/pages/maps/overview/events.md]: > (It just seemed a little light to me.)

That's how light and 💯 Kotlin is 😃 

Added comment into the Java & Kotlin code snippets in that section

**Comment by langsmith** [comment]
Ok @colleenmcginnis , all feedback has been addressed.

File: src/pages/maps/overview/events.md (hunk 6 above)
Comment (colleenmcginnis) [reply src/pages/maps/overview/events.md]: 🎉

**Pull request merged**